ts/session/medium_group/index.ts

```diff
@@ -626,106 +619,115 @@ async function sendGroupUpdateForExistingMembers(
     members: membersBin,
     groupName,
     admins: adminsBin,
-    senderKeys,
   };
 
-  const message = new MediumGroupUpdateMessage(params);
-
-  remainingMembers.forEach(async member => {
-    const memberPubKey = new PubKey(member);
-    await getMessageQueue().sendUsingMultiDevice(memberPubKey, message);
-  });
-
-  // Remove sender keys from the params to send to leaving memebers
-  params.senderKeys = [];
-  const strippedMessage = new MediumGroupUpdateMessage(params);
-  leavingMembers.forEach(async member => {
-    const memberPubKey = new PubKey(member);
-    await getMessageQueue().sendUsingMultiDevice(memberPubKey, strippedMessage);
-  });
-}
-
-async function sendGroupUpdateForJoiningMembers(
-  recipients: Array<string>,
-  groupUpdate: GroupInfo,
-  messageId?: string
-) {
-  const { id: groupId, name, members } = groupUpdate;
-
-  const now = Date.now();
-
-  const { secretKey, senderKeysContainer } = groupUpdate;
-
-  if (!secretKey) {
-    window.log.error('Group secret key not specified, aborting...');
-    return;
-  }
+  if (wasAnyUserRemoved) {
+    if (isUserLeaving && leavingMembers.length !== 1) {
+      window.log.warn("Can't remove self and others simultaneously.");
+      return;
+    }
+    // Send the update to the group (don't include new ratchets as everyone should regenerate new ratchets individually)
+    const paramsWithoutSenderKeys = {
+      ...params,
+      senderKeys: [],
+    };
 
-  let senderKeys: Array<RatchetState> = [];
-  if (!senderKeysContainer) {
-    window.log.warn('Sender keys for joining members not found');
-  } else {
-    // Joining members should receive all known sender keys
-    senderKeys = _.union(
-      senderKeysContainer.existingKeys,
-      senderKeysContainer.newKeys
+    const messageStripped = new MediumGroupUpdateMessage(
+      paramsWithoutSenderKeys
     );
-  }
-
-  const membersBin = members.map(
-    (pkHex: string) => new Uint8Array(fromHex(pkHex))
-  );
-
-  const admins = groupUpdate.admins || [];
-
-  const adminsBin = admins.map(
-    (pkHex: string) => new Uint8Array(fromHex(pkHex))
-  );
-
-  const createParams = {
-    timestamp: now,
-    groupId,
-    identifier: messageId || uuid(),
-    groupSecretKey: secretKey,
-    members: membersBin,
-    groupName: name,
-    admins: adminsBin,
-    senderKeys,
-  };
-
-  const mediumGroupCreateMessage = new MediumGroupCreateMessage(createParams);
-
-  recipients.forEach(async member => {
-    const memberPubKey = new PubKey(member);
-    await getMessageQueue().sendUsingMultiDevice(
-      memberPubKey,
-      mediumGroupCreateMessage
+    window.log.warn('Sending to groupUpdateMessage without senderKeys');
+    await getMessageQueue().sendToGroup(messageStripped);
+
+    getMessageQueue().events.addListener('success', async message => {
```

Comment:
Just be careful that the message might not be sent because a user closed the application. Thus the next time they open the application they'll send the message but this might not be triggered